src/controllers/auth.controller.js

```diff
@@ -0,0 +1,325 @@
+import {
+  usersService,
+  emailService,
+  jwtService,
+  tokenService,
+} from './../services/index.js';
+import { userSchema } from './../schemas/index.js';
+import { ApiError } from '../exeptions/api.error.js';
+import { dynamicSchema } from './../utils/dynamicSchema.js';
+import bcrypt from 'bcrypt';
+
+const register = async (req, res) => {
+  const userInformation = req.body;
+
+  const { error, value } = userSchema.validate(userInformation, {
+    allowUnknown: true,
+  });
+
+  const hashedPassword = await bcrypt.hash(value.password, 10);
+
+  if (error) {
+    throw ApiError.badRequest(error.message, { field: error.message });
+  }
+
+  await usersService.register({ ...value, password: hashedPassword });
+
+  res.send({ message: 'OK' });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+
+  const user = await usersService.getOne({ activationToken });
+
+  if (!user) {
+    throw ApiError.notFound({ user: 'User not found' });
+  }
+
+  await usersService.update({ activationToken }, { activationToken: null });
+
+  res.send({ message: 'OK' });
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await usersService.getOne({ email });
+
+  if (!user) {
+    throw ApiError.unauthorized({ user: 'invalid email or password' });
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.unauthorized({ user: 'invalid email or password' });
+  }
+
+  if (user.activationToken) {
+    throw ApiError.unauthorized({ user: 'profile is not activated' });
+  }
+
+  generateTokens(res, user);
```

Comment:
The `generateTokens` function is called here, but it is defined later in the file. Consider moving the function definition above its first usage for better readability.